### PR TITLE
feat: add script for components images update

### PR DESCRIPTION
**Pull request by kkujawa-sumo**
example usage:
```
make update-components-images IMAGES_FILE=/operator/scripts/images.txt 
```

example images.txt:
```
./scripts/get_images_sha256.sh
registry.connect.redhat.com/sumologic/fluent-bit:3.0.6-ubi
MISSING
registry.connect.redhat.com/sumologic/autoinstrumentation-dotnet:0.7.0-ubi
MISSING
registry.connect.redhat.com/sumologic/autoinstrumentation-java:1.26.0-ubi
MISSING
registry.connect.redhat.com/sumologic/autoinstrumentation-nodejs:0.40.0-ubi
MISSING
registry.connect.redhat.com/sumologic/autoinstrumentation-python:0.39b0-ubi
MISSING
registry.connect.redhat.com/sumologic/busybox:1.36.0-ubi
MISSING
registry.connect.redhat.com/sumologic/busybox:latest-ubi
MISSING
registry.connect.redhat.com/sumologic/kube-rbac-proxy:v0.11.0-ubi
MISSING
registry.connect.redhat.com/sumologic/kube-rbac-proxy:v0.15.0-ubi
registry.connect.redhat.com/sumologic/kube-rbac-proxy:@sha256:1153a4592746b05e019bde4d818d176ff9350c013f84d49829032540de882841
registry.connect.redhat.com/sumologic/kube-state-metrics:v2.7.0-ubi
MISSING
registry.connect.redhat.com/sumologic/metrics-server:0.7.0-debian-12-r8-ubi
MISSING
registry.connect.redhat.com/sumologic/nginx-unprivileged:1.26.0-alpine-sumo-0-ubi
MISSING
registry.connect.redhat.com/sumologic/node-exporter:v1.3.1-ubi
registry.connect.redhat.com/sumologic/node-exporter:@sha256:7acec4473ddf508514dca1d08335cfd071e345d7eca660793d59e09ef9f0491f
registry.connect.redhat.com/sumologic/opentelemetry-operator:0.95.0-ubi
MISSING
registry.connect.redhat.com/sumologic/prometheus-config-reloader:v0.59.2-ubi
MISSING
registry.connect.redhat.com/sumologic/prometheus-operator:v0.59.2-ubi
MISSING
registry.connect.redhat.com/sumologic/prometheus:v2.39.0-ubi
MISSING
registry.connect.redhat.com/sumologic/telegraf-operator:v1.3.11-ubi
MISSING
registry.connect.redhat.com/sumologic/telegraf:1.21.2-ubi
registry.connect.redhat.com/sumologic/telegraf:@sha256:ca396dad12a289aea9136da713020d31b179e9f49aae61c48332d61086d1d059
registry.connect.redhat.com/sumologic/thanos:v0.28.0-ubi
MISSING
```

Documentation will be added in next pull request.

### CODE REVIEW

File: scripts/update_images.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+RED_HAT_REGISTRY = "registry.connect.redhat.com/sumologic/"
+ENV_PREFIX = "RELATED_IMAGE_"
+CLUSTER_SERVICE_VERSION_PATH = "bundle/manifests/operator.clusterserviceversion.yaml"
+MANAGER_PATH = "config/manager/manager.yaml"
+
+def remove_prefix(s: str, prefix: str) -> str:
+    if s.startswith(prefix):
+        return s[len(prefix):]
+    else:
+        return s
```
Comment (sumo-drosiek) [inline scripts/update_images.py:15]: https://docs.python.org/3/library/stdtypes.html#str.removeprefix 🤔

**Comment by sumo-drosiek** [comment]
Could you add some comments to the function? It will help with maintaining in the future

File: scripts/update_images.py (hunk 1 above)
Comment (kkujawa-sumo) [reply scripts/update_images.py]: Changed :)

File: scripts/update_images.py [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+     Returns:
+        list: list of related images
+        list: list of environment variables with information about components images
+    """
+    lines = get_lines(image_list_file)
+    related_images = []
+    image_envs = []
+    for image_with_tag, image_with_sha256 in pairwise(lines):
+        component_and_tag =image_with_tag.removeprefix(RED_HAT_REGISTRY).split(":")
```
Comment (sumo-drosiek) [inline scripts/update_images.py:56]:
```suggestion
        component_and_tag = image_with_tag.removeprefix(RED_HAT_REGISTRY).split(":")
```

File: scripts/update_images.py [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+        helm_operator_image = get_helm_operator_image(related_images)
+        new_related_images.insert(0, helm_operator_image)
+        cluster_service_version["spec"]["relatedImages"] = new_related_images
+
+        # todo: search for the helm operator container
+        for i in range(len(cluster_service_version["spec"]["install"]["spec"]["deployments"][0]["spec"]["template"]["spec"]["containers"])):
+            name = cluster_service_version["spec"]["install"]["spec"]["deployments"][0]["spec"]["template"]["spec"]["containers"][i]["name"]
+            if name == "operator":
+                envs = cluster_service_version["spec"]["install"]["spec"]["deployments"][0]["spec"]["template"]["spec"]["containers"][i]["env"]
+                cluster_service_version["spec"]["install"]["spec"]["deployments"][0]["spec"]["template"]["spec"]["containers"][i]["env"] = update_envs(envs, new_image_envs)
```
Comment (sumo-drosiek) [inline scripts/update_images.py:134]: I would assign `cluster_service_version["spec"]["install"]["spec"]["deployments"][0]["spec"]["template"]["spec"]["containers"]` to variable to increase readability

File: scripts/update_images.py [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+        list: list of related images
+        list: list of environment variables with information about components images
+    """
+    lines = get_lines(image_list_file)
+    related_images = []
+    image_envs = []
+    for image_with_tag, image_with_sha256 in pairwise(lines):
+        component_and_tag = image_with_tag.removeprefix(RED_HAT_REGISTRY).split(":")
+        component = component_and_tag[0]
+        tag = component_and_tag[1]
```
Comment (sumo-drosiek) [inline scripts/update_images.py:64]:
```suggestion
        component, tag = component_and_tag[0]
        tag = component_and_tag[1]
```
```suggestion
        component, tag = image_with_tag.removeprefix(RED_HAT_REGISTRY).split(":")
```

**Review by sumo-drosiek — APPROVED** [review]
One last suggestion and LGTM

**Pull request merged**